src/app/utils/bibDetailsUtils.jsx

```diff
@@ -29,4 +29,69 @@ const combineBibDetailsData = (bibDetails, additionalData) => {
   return bibDetails.concat(filteredAdditionalData);
 };
 
-export { definitionItem, annotatedMarcDetails, combineBibDetailsData };
+
+/**
+ * isRtl(string)
+ * Returns true (or false) depending on whether the string needs to be read right to left
+ * @param {string} string
+ * @return  {boolean}
+ */
+const isRtl  = (string) => {
+  if (typeof string !== 'string') { return null }
+  return string.substring(0, 1) === '\u200F'
+}
+
+/**
+ * stringDirection(string)
+ * 'rtl' if the string needs to be read right to left, otherwise null
+ * @param {string} string
+ * @return {string}
+ */
+const stringDirection = (string) => {
+  return isRtl(string) ? 'rtl' : null
+}
+
+
+/**
+ * interleave(arr1, arr2)
+ * Given two arrays, returns the elements interleaved, with falsey elements removed.
+ * Example: interleave ([1, 2, null, 3], [5,6,7,8,9]) =>
+ * [1,5,2,6,7,3,8,9].
+ * Assumes that arr2 is at least as long as arr1.
+ *
+ * @param {array} arr1
+ * @param {array} arr2
+ * @return {array}
+ */
+const interleave = (arr1, arr2) => arr2.reduce((acc, el, id) => (arr1[id] && acc.push(arr1[id]), el && acc.push(el), acc), [])
+
+
+/**
+ * matchParallels(bib)
+ * Given a bib object returns a new copy of the bib in which fields with parallels have been rewritten
+ * The new rewritten field interleaves the parallel field and the paralleled (i.e. original) field together.
+ * Skips over subject fields since these require changes to SHEP
+ * @param {object} bib
+ * @return {object}
+ */
+const matchParallels = (bib) => {
+  const parallelFieldMatches = Object.keys(bib).map((key) => {
```

Comment:
Thinking still about feature flagging this work: Could we effectively control display of parallels by adding a feature flag check here? (e.g. on line 78: `if (!appConfig.features.includes('parallels')) return bib`) May also need to add it to [the parallelTitle replacement bit](https://github.com/NYPL/discovery-front-end/pull/1915/files#diff-7ee04a7cd65f20ba576a44c1c7d827390839d2f50da64b6c75e9b3e04969c40cR93) Seems like including the `@dir` code and related css is harmless if we disable the matching work in these two places? Would be nice to get this work into `development` before more structural changes